Sources/SwiftParser/ExpressionInterpretedAsVersionTuple.swift

```diff
@@ -0,0 +1,21 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the Swift.org open source project
+//
+// Copyright (c) 2014 - 2023 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See https://swift.org/LICENSE.txt for license information
+// See https://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+//
+//===----------------------------------------------------------------------===//
+
+@_spi(RawSyntax) import SwiftSyntax
+
+extension ExprSyntax {
+  public func asVersionTuple(with maxComponentCount: Int) -> VersionTupleSyntax? {
```

Comment:
Could you add a doc comment explaining what this does and why it’s necessary?

---

`asVersionTuple` sound like a cast to me. I would prefer something like `interpretAsVersionTuple`.